app/src/main/java/io/stormbird/wallet/entity/tokenscript/TokenscriptFunction.java

```diff
@@ -43,6 +47,209 @@ public Function generateTransactionFunction(String walletAddr, BigInteger tokenI
             resolveReference(arg, tokenId, definition, attrIf);
             switch (arg.parameterType)
             {
+                case "e18":
```

Comment:
I don't think e18 should go in here - it should be handled before we get here probably. It should be handled more programmatically anyway (my fault probably) that is the number should be read and the input scaled accordingly.

Reply:
It seems sangalli has moved on, can you fix this after him?

Reply:
> that is the number should be read and the input scaled accordingly.

If we decide to do this, can we write it in the doc and create an issue on iOS to do that too? Right now, "e18" is considered as as special case, we don't handle arbitrary "eN", where N is a whole number.

Reply:
@hboon @colourful-land we should handle eN because e18 is not the only possibility

Reply:
I proposed before and I do it again: I propose only implement a few (no more than 4) types of e, like e8, e18, e10 etc.

The reason being if you want to generalise it you may as well generalise it properly, e.g. why not raise the number to base 2? Let's say we use b24 to represent that the value has to be bit-shifted 24 positions.

Reply:
I can put in some popular e multipliers into the new schema and change this as we move to october schema.